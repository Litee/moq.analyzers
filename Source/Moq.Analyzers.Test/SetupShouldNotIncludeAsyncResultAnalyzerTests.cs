namespace Moq.Analyzers.Test
{
    using System.IO;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using Microsoft.CodeAnalysis.Diagnostics;
    using TestHelper;
    using Xunit;

    [UseReporter(typeof(DiffReporter))]
    public class SetupShouldNotIncludeAsyncResultAnalyzerTests : DiagnosticVerifier
    {
        [Fact]
        public void ShouldPassIfSetupProperly()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/SetupShouldNotIncludeAsyncResult.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new SetupShouldNotIncludeAsyncResultAnalyzer();
        }
    }
}
