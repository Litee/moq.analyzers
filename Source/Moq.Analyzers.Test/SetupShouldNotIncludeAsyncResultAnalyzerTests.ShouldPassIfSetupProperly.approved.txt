Diagnostic 1
	Id: Moq1201
	Location: SourceFile(Test0.cs[871..912))
	Highlight: c.GenericAsyncWithConcreteReturn().Result
	Lines: mock.Setup(c => c.GenericAsyncWithConcreteReturn().Result);
	Severity: Error
	Message: Setup of async methods should use ReturnsAsync instead of .Result

